developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,571 @@
+
+Depending on whether the extension module is required to get a functioning
+interpreter or not, we update :cpy-file:`Modules/Setup.bootstrap.in` or
+:cpy-file:`Modules/Setup.stdlib.in`. In the former case, the extension
+module is necessarily built as a built-in extension module.
+
+.. tip::
+
+   For accelerator modules, :cpy-file:`Modules/Setup.stdlib.in` should be
+   preferred over :cpy-file:`Modules/Setup.bootstrap.in`.
+
+For built-in extension modules, update :cpy-file:`Modules/Setup.bootstrap.in`
+by adding the following line after the ``*static*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.bootstrap.in`
+   :emphasize-lines: 3
+
+   *static*
+   ...
+   fastfoo _foo/_foomodule.c _foo/helper.c
+   ...
+
+For other extension modules, update :cpy-file:`Modules/Setup.stdlib.in`
+by adding the following line after the ``*@MODULE_BUILDTYPE@*`` marker
+but before the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 3
+
+   *@MODULE_BUILDTYPE@*
+   ...
+   @MODULE_FASTFOO_TRUE@fastfoo _foo/_foomodule.c _foo/helper.c
+   ...
+   *shared*
+
+The ``@MODULE_<NAME>_TRUE@<name>`` marker expects ``<NAME>`` to be the
+upper-cased module name ``<name>``. If the extension module must be built
+as a *shared* module, put the ``@MODULE_FASTFOO_TRUE@fastfoo`` line after
+the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 4
+
+   ...
+   *shared*
+   ...
+   @MODULE_FASTFOO_TRUE@fastfoo _foo/_foomodule.c _foo/helper.c
+
+Compiling the CPython project
+-----------------------------
+
+Now that everything is in place, it remains to compile the project:
+
+.. code-block:: shell
+
+   make regen-configure
+   ./configure
+   make regen-all
+   make regen-stdlib-module-names
+   make
+
+.. tip::
+
+   Use ``make -j12`` to speed-up compilation if you have enough CPU cores.
+
+* ``make regen-configure`` updates the :cpy-file:`configure` script.
+
+* ``make regen-all`` is responsible for regenerating header files and
+  invoking other scripts, such as :ref:`Argument Clinic <clinic>`.
+  Execute this rule if you do not know which files should be updated.
+
+* ``make regen-stdlib-module-names`` updates the standard module names, making
+  :mod:`!fastfoo` discoverable and importable via ``import fastfoo``.
+
+* The final ``make`` step is generally not needed since the previous ``make``
+  invokations may completely rebuild the project, but it could be needed in
+  some specific cases.
+
+Troubleshooting
+---------------
+
+This section addresses common issues that you may face when following this tutorial.
+
+No rule to make target ``regen-configure``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This usually happens after running ``make distclean`` (which removes
+the ``Makefile``). The solution is to regenerate the :cpy-file:`configure`
+script as follows:
```

Comment:
Or run `Tools/build/regen-configure.sh` directly.

Reply:
I thought about that, but I wanted to reduce the number of tools or rules I introduce. Also, if by any chance the Makefile is doing something else, at least, I wouldn't need to update that section. But I can mention it if you think it's better.